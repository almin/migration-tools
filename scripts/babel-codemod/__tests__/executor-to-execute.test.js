"use strict";
const childProcess = require('child_process');
const { promisify } = require('util');
const path = require('path');
const fs = require('fs');
const exec = promisify(childProcess.exec)
const codemod = require.resolve(".bin/codemod");

const run = (pluginFilePath, filePath) => {
    if (!fs.existsSync(pluginFilePath)) {
        return Promise.reject(new Error(`Not exist ${pluginFilePath}`));
    }
    if (!fs.existsSync(filePath)) {
        return Promise.reject(new Error(`Not exist ${filePath}`));
    }
    // TODO: https://github.com/square/babel-codemod/issues/13
    return exec(`cat "${filePath}" | ${codemod} --plugin "${pluginFilePath}" --stdio`).then(result => {
        // TODO: IT IS DEBUG HACK
        if (result.stderr) {
            console.log(result.stderr);
        }
        return result.stdout;
    })
};

const testFile = (testName) => {
    it(testName, async () => {
        const filePath = path.join(__dirname, `../__testfixtures__/executor-to-execute/${testName}`);
        const inputContent = fs.readFileSync(filePath, "utf-8");
        const stdout = await run(
            path.join(__dirname, "../executor-to-execute.js"),
            filePath,
        );
        expect(`${inputContent}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
${stdout}`).toMatchSnapshot();
    });
};
jest.setTimeout(20000);
describe("executor-to-execute", () => {
    testFile("simple.js");
    testFile("simple.ts");
    testFile("evincive-return-execute.ts");
    testFile("return-function-expression.ts");
    testFile("no-arguments.ts");
    testFile("does-not-change-executor.ts");
});
