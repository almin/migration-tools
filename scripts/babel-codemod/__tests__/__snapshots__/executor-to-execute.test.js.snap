// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`executor-to-execute does-not-change-executor 1`] = `
"const expression = useCase => useCase.execute(\\"A\\");
context.useCase(new MyUseCaseA()).executor(expression);

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const expression = useCase => useCase.execute(\\"A\\");
context.useCase(new MyUseCaseA()).executor(expression);
"
`;

exports[`executor-to-execute does-not-change-executor.ts 1`] = `
"const expression = useCase => useCase.execute(\\"A\\");
context.useCase(new MyUseCaseA()).executor(expression);

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const expression = useCase => useCase.execute(\\"A\\");
context.useCase(new MyUseCaseA()).executor(expression);
"
`;

exports[`executor-to-execute evincive-return-execute 1`] = `
"context.useCase(createMyUseCaseA()).executor(useCase => {
    return useCase.execute(\\"A\\")
});

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
context.useCase(createMyUseCaseA()).execute(\\"A\\");
"
`;

exports[`executor-to-execute evincive-return-execute.ts 1`] = `
"const value: string = \\"asasea\\";
context.useCase(createMyUseCaseA()).executor(useCase => {
    return useCase.execute(value)
});

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const value: string = \\"asasea\\";
context.useCase(createMyUseCaseA()).execute(value);
"
`;

exports[`executor-to-execute no-arguments 1`] = `
"context.useCase(new MyUseCaseA()).executor(useCase => useCase.execute());

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
context.useCase(new MyUseCaseA()).execute();
"
`;

exports[`executor-to-execute no-arguments.ts 1`] = `
"context.useCase(new MyUseCaseA()).executor(useCase => useCase.execute());

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
context.useCase(new MyUseCaseA()).execute();
"
`;

exports[`executor-to-execute return-function-expression 1`] = `
"context.useCase(createMyUseCaseA()).executor(function (useCase) {
    return useCase.execute(1, 2, { key: \\"3\\" })
});

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
context.useCase(createMyUseCaseA()).execute(1, 2, { key: \\"3\\" });
"
`;

exports[`executor-to-execute return-function-expression.ts 1`] = `
"context.useCase(createMyUseCaseA()).executor(function (useCase) {
    return useCase.execute(1, 2, { key: \\"3\\" })
});

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
context.useCase(createMyUseCaseA()).execute(1, 2, { key: \\"3\\" });
"
`;

exports[`executor-to-execute simple 1`] = `
"import { UseCase, Context } from \\"almin\\";

class MyUseCaseA extends UseCase {
    execute(_a: string) {
    }
}

const context = new Context({
    store: createStore({ name: \\"test\\" })
});

context.useCase(new MyUseCaseA()).executor(useCase => useCase.execute(\\"A\\"));

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import { UseCase, Context } from \\"almin\\";

class MyUseCaseA extends UseCase {
    execute(_a: string) {
    }
}

const context = new Context({
    store: createStore({ name: \\"test\\" })
});

context.useCase(new MyUseCaseA()).execute(\\"A\\");
"
`;

exports[`executor-to-execute simple.js 1`] = `
"import { UseCase, Context } from \\"almin\\";

class MyUseCaseA extends UseCase {
    execute(_a) {
        console.log(_a)
    }
}

const context = new Context({
    store: createStore({ name: \\"test\\" })
});

context.useCase(new MyUseCaseA()).executor(useCase => useCase.execute(\\"A\\"));

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import { UseCase, Context } from \\"almin\\";

class MyUseCaseA extends UseCase {
    execute(_a) {
        console.log(_a)
    }
}

const context = new Context({
    store: createStore({ name: \\"test\\" })
});

context.useCase(new MyUseCaseA()).execute(\\"A\\");
"
`;

exports[`executor-to-execute simple.ts 1`] = `
"import { UseCase, Context } from \\"almin\\";

class MyUseCaseA extends UseCase {
    execute(_a: string) {
    }
}

const context = new Context({
    store: createStore({ name: \\"test\\" })
});

context.useCase(new MyUseCaseA()).executor(useCase => useCase.execute(\\"A\\"));

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
import { UseCase, Context } from \\"almin\\";

class MyUseCaseA extends UseCase {
    execute(_a: string) {
    }
}

const context = new Context({
    store: createStore({ name: \\"test\\" })
});

context.useCase(new MyUseCaseA()).execute(\\"A\\");
"
`;
